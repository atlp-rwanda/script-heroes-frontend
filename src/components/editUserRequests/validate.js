const validate = (state) => {
  let errors = {};
  if (!state.location) {
    errors.locationError = 'Location is required';
  }
  if (!state.destination) {
    errors.destinationError = 'Destination is required';
  }
  if (!state.start) {
    errors.startError = 'Start Date is required';
  }
  if (!state.end) {
    errors.endError = 'End Date is required';
  }
  if (!state.reason) {
    errors.reasonError = 'Travel Reason is required';
  }
  if (!state.accomodation) {
    errors.accomodationError = 'Accommodation is required';
  }
  return errors;
};
export default validate;
