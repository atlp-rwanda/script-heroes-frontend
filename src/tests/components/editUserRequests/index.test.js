import React from 'react';
import { mount } from 'enzyme';
import { Form, Input } from 'reactstrap';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import UpdateRequest from '../../../components/editUserRequests';
import validate from '../../../components/editUserRequests/validate';

describe('<UpdateRequests/>', () => {
  const initialState = {
    locations: { locations: [] },
    accommodations: { accommodations: [] },
  };
  const middlewares = [thunk];
  const mockStore = configureStore(middlewares);
  let store;
  let wrapper;
  let preventDefault = jest.fn();
  let state = [
    { name: 'location', value: 'Kigali' },
    { name: 'destination', value: 'Kigali' },
    { name: 'start', value: '01-01-2021' },
    { name: 'end', value: '10-01-2021' },
    { name: 'reason', value: 'Vacation' },
    { name: 'accomodation', value: 'Mariot Hotel' },
  ];
  let testState = {};
  let newState = {
    location: 'Kigali',
    destination: 'Kenya',
    start: '01-01-2021',
    end: '10-01-2021',
    reason: 'Vacation',
    accomodation: 'Mariot Hotel',
  };
  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = mount(
      <Provider store={store}>
        <Router>
          <UpdateRequest />
        </Router>
      </Provider>
    );
  });
  it('Should render without errors', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('Should not submit invalid input', () => {
    wrapper.find(Form).at(0).simulate('submit', { preventDefault });
    expect(preventDefault).toBeCalled();
  });
  it('Should test on change', () => {
    let inputs = wrapper.find(Input);
    for (let index = 0; index < inputs.length; index++) {
      inputs.at(index).simulate('change', {
        target: {
          name: state[index].name,
          value: state[index].value,
        },
      });
    }
    wrapper.find(Form).at(0).simulate('submit', { preventDefault });
    expect(preventDefault).toBeCalled();
  });

  it('Should test validation function', () => {
    expect(validate(testState)).toStrictEqual({
      accomodationError: 'Accommodation is required',
      destinationError: 'Destination is required',
      endError: 'End Date is required',
      locationError: 'Location is required',
      reasonError: 'Travel Reason is required',
      startError: 'Start Date is required',
    });
  });
  it('Should test validation function with no errors', () => {
    expect(validate(newState)).toStrictEqual({});
  });
});
