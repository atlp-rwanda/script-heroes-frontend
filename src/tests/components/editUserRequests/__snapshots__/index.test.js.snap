// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpdateRequests/> Should render without errors 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <UpdateRequest>
        <Container
          className="mb-5 p-5"
          tag="div"
        >
          <div
            className="mb-5 p-5 container"
          >
            <Form
              className="mt-5"
              onSubmit={[Function]}
              tag="form"
            >
              <form
                className="mt-5"
                onSubmit={[Function]}
              >
                <h3
                  className="text-center mt-3 text-color-new"
                >
                  Update Request
                </h3>
                <Row
                  className="mt-5 background-new"
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                >
                  <div
                    className="mt-5 background-new row"
                  >
                    <Col
                      md="2"
                      tag="div"
                      widths={
                        Array [
                          "xs",
                          "sm",
                          "md",
                          "lg",
                          "xl",
                        ]
                      }
                    >
                      <div
                        className="col-md-2"
                      >
                        <FormGroup
                          tag="div"
                        >
                          <div
                            className="form-group"
                          >
                            <Label
                              tag="label"
                              widths={
                                Array [
                                  "xs",
                                  "sm",
                                  "md",
                                  "lg",
                                  "xl",
                                ]
                              }
                            >
                              <label
                                className=""
                              >
                                Location
                              </label>
                            </Label>
                            <Input
                              className="border-success"
                              name="location"
                              onChange={[Function]}
                              type="select"
                              value=""
                            >
                              <select
                                className="border-success form-control"
                                name="location"
                                onChange={[Function]}
                                value=""
                              />
                            </Input>
                          </div>
                        </FormGroup>
                      </div>
                    </Col>
                    <Col
                      md="2"
                      tag="div"
                      widths={
                        Array [
                          "xs",
                          "sm",
                          "md",
                          "lg",
                          "xl",
                        ]
                      }
                    >
                      <div
                        className="col-md-2"
                      >
                        <FormGroup
                          tag="div"
                        >
                          <div
                            className="form-group"
                          >
                            <Label
                              tag="label"
                              widths={
                                Array [
                                  "xs",
                                  "sm",
                                  "md",
                                  "lg",
                                  "xl",
                                ]
                              }
                            >
                              <label
                                className=""
                              >
                                Destination
                              </label>
                            </Label>
                            <Input
                              className="border-success"
                              name="destination"
                              onChange={[Function]}
                              type="select"
                              value=""
                            >
                              <select
                                className="border-success form-control"
                                name="destination"
                                onChange={[Function]}
                                value=""
                              />
                            </Input>
                          </div>
                        </FormGroup>
                      </div>
                    </Col>
                    <Col
                      md="2"
                      tag="div"
                      widths={
                        Array [
                          "xs",
                          "sm",
                          "md",
                          "lg",
                          "xl",
                        ]
                      }
                    >
                      <div
                        className="col-md-2"
                      >
                        <FormGroup
                          tag="div"
                        >
                          <div
                            className="form-group"
                          >
                            <Label
                              tag="label"
                              widths={
                                Array [
                                  "xs",
                                  "sm",
                                  "md",
                                  "lg",
                                  "xl",
                                ]
                              }
                            >
                              <label
                                className=""
                              >
                                Start Date
                              </label>
                            </Label>
                            <Input
                              className="border-success"
                              name="start"
                              onChange={[Function]}
                              type="date"
                              value=""
                            >
                              <input
                                className="border-success form-control"
                                name="start"
                                onChange={[Function]}
                                type="date"
                                value=""
                              />
                            </Input>
                          </div>
                        </FormGroup>
                      </div>
                    </Col>
                    <Col
                      md="2"
                      tag="div"
                      widths={
                        Array [
                          "xs",
                          "sm",
                          "md",
                          "lg",
                          "xl",
                        ]
                      }
                    >
                      <div
                        className="col-md-2"
                      >
                        <FormGroup
                          tag="div"
                        >
                          <div
                            className="form-group"
                          >
                            <Label
                              tag="label"
                              widths={
                                Array [
                                  "xs",
                                  "sm",
                                  "md",
                                  "lg",
                                  "xl",
                                ]
                              }
                            >
                              <label
                                className=""
                              >
                                Return Date
                              </label>
                            </Label>
                            <Input
                              className="border-success"
                              name="end"
                              onChange={[Function]}
                              type="date"
                              value=""
                            >
                              <input
                                className="border-success form-control"
                                name="end"
                                onChange={[Function]}
                                type="date"
                                value=""
                              />
                            </Input>
                          </div>
                        </FormGroup>
                      </div>
                    </Col>
                    <Col
                      md="2"
                      tag="div"
                      widths={
                        Array [
                          "xs",
                          "sm",
                          "md",
                          "lg",
                          "xl",
                        ]
                      }
                    >
                      <div
                        className="col-md-2"
                      >
                        <FormGroup
                          tag="div"
                        >
                          <div
                            className="form-group"
                          >
                            <Label
                              tag="label"
                              widths={
                                Array [
                                  "xs",
                                  "sm",
                                  "md",
                                  "lg",
                                  "xl",
                                ]
                              }
                            >
                              <label
                                className=""
                              >
                                Travel Reasons
                              </label>
                            </Label>
                            <Input
                              className="border-success"
                              name="reason"
                              onChange={[Function]}
                              placeholder="eg: Trip"
                              type="text"
                              value=""
                            >
                              <input
                                className="border-success form-control"
                                name="reason"
                                onChange={[Function]}
                                placeholder="eg: Trip"
                                type="text"
                                value=""
                              />
                            </Input>
                          </div>
                        </FormGroup>
                      </div>
                    </Col>
                    <Col
                      md="2"
                      tag="div"
                      widths={
                        Array [
                          "xs",
                          "sm",
                          "md",
                          "lg",
                          "xl",
                        ]
                      }
                    >
                      <div
                        className="col-md-2"
                      >
                        <FormGroup
                          tag="div"
                        >
                          <div
                            className="form-group"
                          >
                            <Label
                              tag="label"
                              widths={
                                Array [
                                  "xs",
                                  "sm",
                                  "md",
                                  "lg",
                                  "xl",
                                ]
                              }
                            >
                              <label
                                className=""
                              >
                                Accommodation
                              </label>
                            </Label>
                            <Input
                              className="border-success"
                              name="accomodation"
                              onChange={[Function]}
                              type="select"
                              value=""
                            >
                              <select
                                className="border-success form-control"
                                name="accomodation"
                                onChange={[Function]}
                                value=""
                              />
                            </Input>
                          </div>
                        </FormGroup>
                      </div>
                    </Col>
                  </div>
                </Row>
                <div
                  className=" text-center mt-3 mb-3"
                >
                  <Button
                    className="btn-color"
                    color="secondary"
                    tag="button"
                  >
                    <button
                      aria-label={null}
                      className="btn-color btn btn-secondary"
                      onClick={[Function]}
                    >
                      SEND
                    </button>
                  </Button>
                </div>
              </form>
            </Form>
          </div>
        </Container>
      </UpdateRequest>
    </Router>
  </BrowserRouter>
</Provider>
`;
